packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx

```diff
@@ -677,15 +677,15 @@ export class BaseButton extends BaseComponent<IBaseButtonProps, IBaseButtonState
     }
 
     if (!(ev.altKey || ev.metaKey) && (isUp || isDown)) {
-      this.setState(state => {
-        if (state.menuProps && !state.menuProps.shouldFocusOnMount) {
-          return { menuProps: { ...state.menuProps, shouldFocusOnMount: true } };
-        }
-        return state;
-      });
-
-      ev.preventDefault();
-      ev.stopPropagation();
+      const state = this.state;
+      // This should be done in the setStateCallback but because preventDefault
+      // needs to be called, we have to evaluate the current state, even though
+      // it might not be 100% accurate;
+      if (state.menuProps && !state.menuProps.shouldFocusOnMount) {
+        ev.preventDefault();
+        ev.stopPropagation();
+        this.setState({ menuProps: { ...state.menuProps, shouldFocusOnMount: true } });
```

Comment:
Checking state directly for acting is consistent, but might want to use the `setState((state) =>{ menuProps: ...state.menuProps, shouldFocusOnMount: true } }` to make sure we don't overwrite any pending state writes.

Reply:
I agree with Jason, once that's changed I'll approve the PR 😄

Reply:
My only concern with that is that there can then be disconnect between preventing default and the current state, although that's pretty unlikely. I'll make the change.